fusequery/query/src/planners/plan_scheduler.rs

```diff
@@ -110,21 +108,26 @@ impl PlanScheduler {
                 );
                 num_chunks_so_far += chunk_size;
 
-                let mut rewritten_node = PlanNode::Empty(EmptyPlan {
-                    schema: Arc::new(DataSchema::empty())
-                });
+                struct ReadSourceRewriter {
```

Comment:
Can we move this block to out, seems better to read?

Reply:
That's fine.

Reply:
Done.